### ISSUE TITLE
feat: encrypted blocks

### ISSUE THREAD

**Issue by tangjeff0** [athensresearch/athens#1065]
- TODO: do more research on use cases. For personal encryption or for sharing? This affects the kind of encryption (private-public vs cert).

# Roam/Password Approach

Uses password.

See
- https://www.roamtips.com/home/password-protect-roam-blocks
- https://roamresearch.com/#/app/athensresearch/page/bxlcRpmXW

# Certificate Approach

Uses certificate.

# Implementation

- [ ] choose one of the approaches above
- [ ] filter encrypted blocks from auto-complete, search, unlinked references

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 1000.0 USD (999.7 USD @ $1.0/USD)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/athensresearch/athens/1065/100025608).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/athensresearch/athens/1065/100025608).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.com/invite/83BK5z55yB/'>Gitcoin's Discord</a>
* $1,999,757.40 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 265 years, 6 months from now.
Please review their action plans below:


**1) [em-ak](https://gitcoin.co/em-ak) has started work.**

- research a bit more electron capabilities to handle client certificates
  - but the password based option seems more user friendly and portable for now (no key to manage or to loose)
- implement the functions to encrypt / decrypt the content of a block
- document the approach and ask for feedback about security flaws
- wire the functionality to UI elements for easy interaction
- filter out encrypted blocks

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/athensresearch/athens/1065/100025608).

**Comment by tangjeff0** [comment]
Hi @Em-AK I saw you started working on this. I just talked to our main customer who wanted this feature, and they said that Roam's implementation is not actually all that good.

Their general sentiment is that the more encryption, the better. E2EE (#873) is better than encrypted blocks, which are better than encrypted pages (though pages and blocks are pretty similar).

I would love to hear security-minded perspectives to discuss the various levels of encryption. E2EE, encryption in transit, at rest, public-private, certificate. CC @neotyk, @agentydragon. It may not even be necessary to create UI for this, but if we do, I'd love to have @shanberg in on the discussion.
